authentication/models.py

```diff
@@ -33,9 +38,88 @@ class CustomUser(AbstractBaseUser, PermissionsMixin):
     is_active = models.BooleanField(default=True)
     is_staff = models.BooleanField(default=False)
 
+
     USERNAME_FIELD = "email"
     REQUIRED_FIELDS = ["password", "first_name", "surname", "phone_number"]
     objects = CustomUserManager()
 
     def __str__(self):
         return f"{self.first_name} {self.surname} {self.email}"
+    
+    def get_auth_data(self):
+        return {'email': self.email,
+                'password': self.password,
+                'first_name': self.first_name,
+                'surname': self.surname,
+                'user_id': self.user_id,
+                'is_verified': self.is_verified,
+                'is_superuser': self.is_superuser}
+    
+
+class Companies(models.Model):
+    company_id = models.BigAutoField(primary_key=True, unique=True, auto_created=True)
+    brand = models.CharField(max_length=255, blank=True)
+    is_registered = models.BooleanField(default=False)
+    common_info = models.TextField(blank=True)
+    contact_phone = models.CharField(max_length=255, blank=True)
+    contact_email = models.CharField(max_length=255, blank=True)
+    registration_date = models.DateTimeField(auto_now_add=True)
+    edrpou = models.IntegerField(blank=True, null=True)
+    address = models.TextField(max_length=255, blank=True)
+    product_info = models.TextField(blank=True)
+    startup_idea = models.TextField(blank=True)
+    tags = models.CharField(max_length=255, blank=True)
+
+
+class Positions(models.Model):
+    position_id = models.BigAutoField(primary_key=True, unique=True, auto_created=True)
+    position = models.CharField(max_length=100)
+    position_descr = models.CharField(max_length=500, blank=True)
+
+
+class CompaniesAndUsersRelations(models.Model):
+    relation_id = models.BigAutoField(primary_key=True)
+    user_id = models.ForeignKey(CustomUser, on_delete=models.CASCADE)
+    company_id = models.ForeignKey(Companies, on_delete=models.CASCADE)
+    position = models.ForeignKey(Positions, on_delete=models.CASCADE)
+
+
+class AuthUser(AbstractBaseUser):
+    email = models.EmailField(max_length=100, default=None, unique=True)
+    password = models.CharField(max_length=128, default=None)
+    first_name = models.CharField(max_length=100, default=None)
+    surname = models.CharField(max_length=100, default=None)
+    position = models.CharField(max_length=25, default=None)
+    company_id = models.IntegerField(default=None)
```

Comment:
mixes authentication fields with business logic fields like ```company_id``` and ```position```, which could be separated for clarity

Reply:
AuthUser model serves as blueprint for authenticated user and company information only, it isn't intented for migration. If the user_id and company_id are provided in token, we extract the combined information of user and company he is related in the moment and returns it in AuthUser instance